dpctl/_sycl_platform.pyx

```diff
@@ -366,6 +375,78 @@ cdef class SyclPlatform(_SyclPlatform):
         """
         return DPCTLPlatform_Hash(self._platform_ref)
 
+    def get_devices(self, device_type=device_type_t.all):
+        """
+        Returns the list of :class:`dpctl.SyclDevice` objects associated with
+        :class:`dpctl.SyclPlatform` instance selected based on
+        the given :class:`dpctl.device_type`.
+
+        Args:
+            device_type (optional):
+                A :class:`dpctl.device_type` enum value or a string that
+                specifies a SYCL device type. Currently, accepted values are:
+                "gpu", "cpu", "accelerator", or "all".
+                Default: ``dpctl.device_type.all``.
+
+        Returns:
+            list:
+                A :obj:`list` of :class:`dpctl.SyclDevice` objects
+                that belong to this platform.
+
+        Raises:
+            TypeError:
+                If `device_type` is not a str or :class:`dpctl.device_type`
+                enum.
+            ValueError:
+                If the ``DPCTLPlatform_GetDevices`` call returned
+                ``NULL`` instead of a ``DPCTLDeviceVectorRef`` object.
+        """
+        cdef _device_type DTy = _device_type._ALL_DEVICES
+        cdef DPCTLDeviceVectorRef DVRef = NULL
+        cdef size_t num_devs
+        cdef size_t i
+        cdef DPCTLSyclDeviceRef DRef
+
+        if isinstance(device_type, str):
+            dty_str = device_type.strip().lower()
+            if dty_str == "accelerator":
```

Comment:
It seems `"host"` device type is present in the [sycl-2020](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#appendix.device.descriptors), but cleaned up from dpctl and not considered anymore. Is that expected? Should it be clarified in the PR description?

Reply:
It should be clarified. `"host"` device type is no longer present in SYCL 2020's newer revisions, and was deprecated some time ago.

Reply:
I've updated the PR description to mention this, as well as reference https://github.com/IntelPython/dpctl/pull/1028, where host devices were first removed.